### PR TITLE
Update wallet link to be able to delete connection

### PR DESCRIPTION
**Pull request by alexshoykhet**

### CODE REVIEW

File: android/walletlink/src/main/java/com/coinbase/walletlink/daos/SessionDAO.kt [hunk 1]
```diff
@@ -45,12 +45,34 @@ class SessionDAO(private val store: StoreInterface) {
...
+        sessionId: String,
+        secret: String,
+        version: String?,
+        dappName: String?,
+        dappImageURL: URL?,
+        dappURL: URL?
+    ) = accessLock.withLock {
         val sessions = (store.get(StoreKeys.sessions) ?: emptyArray())
-            .filter { it.id != sessionId && it.url == url }.toMutableList()
+            .filter { it.id != sessionId && it.url != url }.toMutableList()
```
Comment (hishbase) [inline android/walletlink/src/main/java/com/coinbase/walletlink/daos/SessionDAO.kt]: 1. should be an OR `||`
2. nit: move `.toMutableList` to next line

File: android/walletlink/src/main/java/com/coinbase/walletlink/daos/SessionDAO.kt [hunk 2]
```diff
@@ -63,7 +85,7 @@ class SessionDAO(private val store: StoreInterface) {
      */
     fun delete(url: URL, sessionId: String) = accessLock.withLock {
         val sessionIds = (store.get(StoreKeys.sessions) ?: arrayOf())
-            .filter { it.id != sessionId && it.url == url }.toMutableList()
+            .filter { it.id != sessionId || it.url != url }.toMutableList()
```
Comment (hishbase) [inline android/walletlink/src/main/java/com/coinbase/walletlink/daos/SessionDAO.kt]: nit: move .toMutableList to next line

File: android/walletlink/src/main/java/com/coinbase/walletlink/apis/WalletLinkWebSocket.kt [hunk 3]
```diff
@@ -221,6 +230,26 @@ internal class WalletLinkWebSocket(val url: URL) {
...
+                    if (it.keys.filterIsInstance<String>().size != it.size ||
+                        it.values.filterIsInstance<Any>().size != it.size) {
+                        null
+                    } else {
+                        it as? Map<String, Any>
+                    }
+                } ?: return
+
+                val destroyedValue = metadata[ClientMetadataKey.Destroyed.rawValue] as? String
+                if (destroyedValue != Strings.destroySession) return
```
Comment (hishbase) [inline android/walletlink/src/main/java/com/coinbase/walletlink/apis/WalletLinkWebSocket.kt]: Does this not work?

```kotlin
val sessionId = (json["sessionId"] as? String) ?: return
val metadata = (json["metadata"] as? Map<String, Any>) ?: return
val destroyedValue = metadata[ClientMetadataKey.Destroyed.rawValue] as? String

if (destroyedValue != Strings.destroySession) return
```

Comment (alexshoykhet) [reply android/walletlink/src/main/java/com/coinbase/walletlink/apis/WalletLinkWebSocket.kt]: There's a warning in the IDE of an unchecked cast

Comment (alexshoykhet) [reply android/walletlink/src/main/java/com/coinbase/walletlink/apis/WalletLinkWebSocket.kt]: I wrote a test to demonstrate this:
```
    @Test
    fun foo() {
        val fooMap: Map<*, *> = mapOf("asdf" to 1, "fdsa" to 1.toBigInteger())

        val barMap = fooMap as? Map<String, Any>
        assertNotNull(barMap)

        val fooMap2: Map<*, *> = mapOf(1 to 1, "fdsa" to 1.toBigInteger())
        val barMap2: Map<String, Any>? = fooMap2 as? Map<String, Any>
        assertNull(barMap2)
    }
```

In this test `barMap2` is actually not null and still a `Map<Integer, Any>`!

Comment (alexshoykhet) [reply android/walletlink/src/main/java/com/coinbase/walletlink/apis/WalletLinkWebSocket.kt]: Then, barMap2 is actually not usable..  For example doing
```
barMap2?.keys?.forEach { }
``` 

Results in a ClassCastException as `forEach` tries to cast `Integer` to `String`.

Comment (alexshoykhet) [reply android/walletlink/src/main/java/com/coinbase/walletlink/apis/WalletLinkWebSocket.kt]: I moved the implementation to an extension in CBCore - https://github.com/CoinbaseWallet/CBCore/pull/34

**Pull request merged**